packages/core/src/state-management/repository.ts

```diff
@@ -180,17 +181,31 @@ export class Repository {
     opts = { ...DEFAULT_LOAD_OPTS, ...opts }
 
     const [state$, synced] = await this.loadingQ.forStream(streamId).run(async () => {
-      const [streamState$, alreadySynced] = await this._loadGenesis(streamId)
-      if (opts.sync == SyncOptions.PREFER_CACHE && alreadySynced) {
-        return [streamState$, alreadySynced]
-      }
-
-      if (opts.sync == SyncOptions.NEVER_SYNC) {
-        return [await this.stateManager.verifyLoneGenesis(streamState$), alreadySynced]
+      switch (opts.sync) {
+        case SyncOptions.PREFER_CACHE: {
+          const [streamState$, alreadySynced] = await this._loadGenesis(streamId)
+          if (alreadySynced) {
+            return [streamState$, alreadySynced]
+          } else {
+            await this.stateManager.sync(streamState$, opts.syncTimeoutSeconds * 1000)
+            return [await this.stateManager.verifyLoneGenesis(streamState$), true]
+          }
+        }
+        case SyncOptions.NEVER_SYNC: {
+          const [streamState$, alreadySynced] = await this._loadGenesis(streamId)
+          return [await this.stateManager.verifyLoneGenesis(streamState$), alreadySynced]
+        }
+        case SyncOptions.SYNC_ALWAYS: {
+          const [fromNetwork$, fromMemoryOrStore] = await Promise.all([
+            this.fromNetwork(streamId),
+            this.fromMemoryOrStore(streamId),
+          ])
+          await this.stateManager.sync(fromNetwork$, opts.syncTimeoutSeconds * 1000, fromMemoryOrStore?.tip)
```

Comment:
```suggestion
          await this.stateManager.sync(fromNetwork$, opts.syncTimeoutSeconds * 1000)
          if (fromMemoryOrStore?.tip) {
            await this.stateManager.handleTip(fromMemoryOrStore?.tip)
          }
```

Reply:
Suggestion to keep the implementation of `sync` simpler and more focused on a single task